js/src/dht.js

```diff
@@ -47,7 +48,12 @@ module.exports = (common) => {
             (cb) => nodeE.swarm.connect(nodeB.peerId.addresses[0], cb),
             (cb) => nodeD.swarm.connect(nodeC.peerId.addresses[0], cb),
             (cb) => nodeE.swarm.connect(nodeC.peerId.addresses[0], cb),
-            (cb) => nodeD.swarm.connect(nodeE.peerId.addresses[0], cb)
+            (cb) => nodeD.swarm.connect(nodeE.peerId.addresses[0], cb),
+            (cb) => nodeA.id((err, id) => {
+              expect(err).to.not.exist()
+              withGo = id.agentVersion.startsWith('go-ipfs')
+              cb()
+            })
```

Comment:
FYI, `spawnNodesWithId` from above means that you already have `nodeA.peerId.agentVersion`

Reply:
Ah, missed that completely. Coming in a soon PR :) Thank you!